### PR TITLE
Refactor the contact retriever

### LINKED ISSUE #52: Rewrite parts of ContactRetriever.
`ContractRetriever` shouldn't be assigning things to member variables. It should have a function called `getContact` that returns the data for a contact.

### LINKED ISSUE #83: Refactor addContact logic in MainActivity
Move addContact stuff out of MainActivity (maybe ContactReceiver? may need to overhaul the whole system #52)

### PR DESCRIPTION
**Pull request by kangp3**
Closes #83 and closes #52

### CODE REVIEW

File: app/src/main/java/xyz/whereuat/whereuat/MainActivity.java [hunk 1]
```diff
@@ -219,20 +223,37 @@ public void showKeyLocDialog(View view) {
...
     @Override
-    public void onActivityResult(int req_code, int result_code, Intent data) {
-        super.onActivityResult(req_code, result_code, data);
-        ContactRetriever con = new ContactRetriever(req_code, result_code, data, this);
+    public void onActivityResult(int req_code, int result_code, Intent intent) {
+        super.onActivityResult(req_code, result_code, intent);
+
+        switch(req_code) {
+            case(Constants.PHONEBOOK_PICK_REQUEST):
+                if (result_code == Activity.RESULT_OK) {
```
Comment (JuliusAlexanderIV) [inline app/src/main/java/xyz/whereuat/whereuat/MainActivity.java]: can you use `AppCompatActivity.RESULT_OK` here so we don't need another import?

Comment (kangp3) [inline app/src/main/java/xyz/whereuat/whereuat/MainActivity.java]: done

File: app/src/main/java/xyz/whereuat/whereuat/utils/PhonebookUtils.java [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+    /**
+     * Helper to retrieve a contact's information from the phone book activity
+     *
+     * @param data The Intent carrying the data of the phone book selection
+     * @param context Context to get content resolver from
+     * @return Contact object storing the name and phone number of the retrieved contact
+     */
+    public static ContactUtils.Contact getContactFromPhonebook(Intent data, Context context) {
+        // Get the data from the Intent
+        Uri contactData = data.getData();
```
Comment (JuliusAlexanderIV) [inline app/src/main/java/xyz/whereuat/whereuat/utils/PhonebookUtils.java]: `contact_data`

**Comment by JuliusAlexanderIV** [comment]
lgtm

**Pull request merged**